identity/migration/src/main/java/io/camunda/identity/migration/MigrationRunner.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.identity.migration;
+
+import static java.util.Arrays.asList;
+
+import io.camunda.service.AuthorizationServices;
+import io.camunda.service.UserServices;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.AuthorizationRecord;
+import io.camunda.zeebe.protocol.impl.record.value.user.UserRecord;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+@Profile("identity-migration")
+public class MigrationRunner implements ApplicationRunner {
+
+  final UserServices<UserRecord> userService;
+
+  final AuthorizationServices<AuthorizationRecord> authorizationServices;
+
+  final RestTemplate restTemplate;
+
+  public MigrationRunner(
+      final UserServices<UserRecord> userService,
+      final AuthorizationServices<AuthorizationRecord> authorizationServices) {
+    this.userService = userService;
+    this.authorizationServices = authorizationServices;
+    restTemplate = new RestTemplate();
+  }
+
+  @Override
+  public void run(final ApplicationArguments args) throws Exception {
+
+    final String command =
+        args.containsOption("command") ? args.getOptionValues("command").getFirst() : "migrate";
```

Comment:
❓ I know that the other command here is `status` would we just include logging to provide updates on where we are in the flow itself? It would also be worth confirming with distribution and controller if we can provide the app with these command line args in their deployment styles :)

Reply:
Ok, good idea. if they can't handle command line args, we can provide a rest endpoint or a special health indicator to return status and stats.